src/main/scala/firrtl/Utils.scala

```diff
@@ -477,6 +478,11 @@ object Utils extends LazyLogging {
             ilen + get_size(t1x.tpe), jlen + get_size(t2x.tpe))
         }._1
       case (ClockType, ClockType) => if (flip1 == flip2) Seq((0, 0)) else Nil
+      case (AsyncResetType, AsyncResetType) => if (flip1 == flip2) Seq((0, 0)) else Nil
+      case (ResetType, other) =>
+        if (legalResetType(other) && flip1 == Default && flip1 == flip2) Seq((0, 0)) else Nil
```

Comment:
Could use a comment with an example. I think this is very subtle, and it would be good for clarity to explain (e.g. why in this case the flips must be default, but in the other case, they must be flip.

Reply:
Done